src/block/put.js

```diff
@@ -22,16 +23,36 @@ module.exports = (send) => {
       return callback(new Error('block.put accepts only one block'))
     }
 
+    let cid = opts.cid
+
     if (typeof block === 'object' && block.data) {
+      if (block.cid) cid = block.cid
       block = block.data
     }
 
-    sendOneFile(block, {}, (err, result) => {
+    let qs = Object.assign(opts, {
+      'input-enc': 'raw',
+      hashAlg: opts.mhtype || 'sha2-256'
+    })
+
+    if (cid && CID.isCID(cid)) {
```

Comment:
At this point `cid` could be a string, buffer or CID instance ([spec](https://github.com/ipfs/interface-ipfs-core/blob/master/SPEC/BLOCK.md#blockput)). `isCID` only checks if the parameter is a CID instance, so currently the passed `cid` will be ignored if it's not a CID instance!